font/src/darwin.rs

```diff
@@ -40,14 +40,49 @@ use core_text::font_descriptor::SymbolicTraitAccessors;
 use core_text::font_descriptor::{CTFontDescriptor, CTFontOrientation};
 
 use euclid::{Point2D, Rect, Size2D};
+use log::{trace, warn};
 
 use super::{FontDesc, FontKey, GlyphKey, Metrics, RasterizedGlyph};
 
-pub mod byte_order;
 use self::byte_order::extract_rgb;
 use self::byte_order::kCGBitmapByteOrder32Host;
 
-use super::Size;
+pub mod byte_order {
+    //! Constants for bitmap byte order
+    #![allow(non_upper_case_globals)]
+    pub const kCGBitmapByteOrder32Little: u32 = 2 << 12;
+    pub const kCGBitmapByteOrder32Big: u32 = 4 << 12;
+
+    #[cfg(target_endian = "little")]
+    pub const kCGBitmapByteOrder32Host: u32 = kCGBitmapByteOrder32Little;
+
+    #[cfg(target_endian = "big")]
+    pub const kCGBitmapByteOrder32Host: u32 = kCGBitmapByteOrder32Big;
+
+    #[cfg(target_endian = "little")]
+    pub fn extract_rgb(bytes: &[u8]) -> Vec<u8> {
+        let pixels = bytes.len() / 4;
+        let mut rgb = Vec::with_capacity(pixels * 3);
+
+        for i in 0..pixels {
+            let offset = i * 4;
+            rgb.push(bytes[offset + 2]);
+            rgb.push(bytes[offset + 1]);
+            rgb.push(bytes[offset]);
+        }
+        rgb
+    }
+
+    #[cfg(target_endian = "big")]
+    pub fn extract_rgb(bytes: Vec<u8>) -> Vec<u8> {
+        bytes
+            .into_iter()
+            .enumerate()
+            .filter(|&(index, _)| ((index) % 4) != 0)
+            .map(|(_, val)| val)
+            .collect::<Vec<_>>()
+    }
+}
```

Comment:
What's the intention behind moving this into the default module? Usually modules in the same file are not desired except for test modules.

Reply:
My intention was to simplify the pre-2018 mod structuring (folder, ambiguous mod.rs file, meaningfully named submodules), while leaving the module path structure unchanged. I can only guess at the original intent, likely a combination of crate constraints and unrealized future plans that needed the scoping. As I can't test for macos before pushing, I'm reluctant to mess around with it more than necessary.

Reply:
> ambiguous mod.rs file

What do you mean by this? How is a `mod.rs` file ever ambiguous?

Modules should never be in the same file without a good reason for it. That's why it originally was put into a different module, since it's unrelated code that has a nice separation from the primary module.

Reply:
`mod.rs` files require a path in order to be distinguished from one another, this was sufficiently annoying to enough people that a new naming scheme was offered.

I'm pretty surprised about the mod aversion, it seems like the perfect application in this case - grouping a small amount of like code in the one file it is used, a grep of this crate's dependencies shows hundreds of non-test mod blocks. I'll revert the darwin directory.

Reply:
The only reason why you would ever go without a `mod.rs` file is when you have only a single module. We have more than one module here, so there's absolutely no point in trying to cram everything into one massive file.

The `mod.rs` file still has its purpose for structures with more than a single module in one directory.

Reply:
FWIW, I don't think `mod.rs` is ambiguous, however it is somewhat _redundant_, which is why the Rust team has added the new resolution in 2018. If I had to choose myself, I'd be in favor of the old style. The main justification is that it's consistent with `lib.rs`.

For small personal projects I can see using the new style, but I think I agree we shouldn't make this change.